lib/galaxy/tool_util/deps/mulled/invfile.lua

```diff
@@ -101,11 +101,11 @@ inv.task('build')
 if VAR.SINGULARITY ~= '' then
     inv.task('singularity')
         .using(singularity_image)
-        .withHostConfig({binds = {"build:/data", "'" .. singularity_image_dir .. "':/import"}, privileged = true})
+        .withHostConfig({binds = {"build:/data", singularity_image_dir .. ":/import"}, privileged = true})
         .withConfig({entrypoint = {'/bin/sh', '-c'}})
```

Comment:
involucro doesn't normally interpret comma-separated arguments in .run() as individual commands, otherwise all other `.run()` wouldn't work. The problem I think is the entrypoint here. Does just removing this line fix the issue as well?

Reply:
You're right, the custom entrypoint is the problem.
But each `.run` seems to start a separate container, so this still fails with
```
[May 14 20:01:14] INFO Invoke Task [singularity]
[May 14 20:01:14] STEP Run image [quay.io/biocontainers/singularity:2.4.6--0] with command [[mkdir -p /usr/local/var/singularity/mnt/container]]
[May 14 20:01:14] DEBU Creating container [step-b7cb30d1fc]
[May 14 20:01:15] DEBU Created container [f98c0b18bd12 step-b7cb30d1fc], starting it
[May 14 20:01:17] DEBU Container [f98c0b18bd12 step-b7cb30d1fc] started, waiting for completion
[May 14 20:01:19] DEBU Container [f98c0b18bd12 step-b7cb30d1fc] completed with exit code [0] as expected
[May 14 20:01:19] DEBU Container [f98c0b18bd12 step-b7cb30d1fc] removed
[May 14 20:01:19] STEP Run image [quay.io/biocontainers/singularity:2.4.6--0] with command [[singularity build /import/zlib /import/Singularity.def]]
[May 14 20:01:19] DEBU Creating container [step-73125f9336]
[May 14 20:01:20] DEBU Created container [65469caa4e48 step-73125f9336], starting it
[May 14 20:01:21] DEBU Container [65469caa4e48 step-73125f9336] started, waiting for completion
[May 14 20:01:21] SOUT Using container recipe deffile: /import/Singularity.def
[May 14 20:01:21] SOUT Sanitizing environment
[May 14 20:01:21] SERR ERROR  : Target /usr/local/var/singularity/mnt/container doesn't exist
[May 14 20:01:21] SERR ABORT  : Retval = 255
[May 14 20:01:21] SOUT Cleaning up...
```

that's for
```diff
diff --git a/lib/galaxy/tool_util/deps/mulled/invfile.lua b/lib/galaxy/tool_util/deps/mulled/invfile.lua
index db1abe49df..b1f2443821 100644
--- a/lib/galaxy/tool_util/deps/mulled/invfile.lua
+++ b/lib/galaxy/tool_util/deps/mulled/invfile.lua
@@ -102,10 +102,9 @@ if VAR.SINGULARITY ~= '' then
     inv.task('singularity')
         .using(singularity_image)
         .withHostConfig({binds = {"build:/data", singularity_image_dir .. ":/import"}, privileged = true})
-        .withConfig({entrypoint = {'/bin/sh', '-c'}})
-        .run('mkdir -p /usr/local/var/singularity/mnt/container && '
-            .. 'singularity build /import/' .. VAR.SINGULARITY_IMAGE_NAME .. ' /import/Singularity.def && '
-            .. 'chown ' .. VAR.USER_ID .. ' /import/' .. VAR.SINGULARITY_IMAGE_NAME)
+        .run('mkdir', '-p', '/usr/local/var/singularity/mnt/container')
+        .run('singularity', 'build', '/import/' .. VAR.SINGULARITY_IMAGE_NAME, '/import/Singularity.def')
+        .run('chown', VAR.USER_ID, '/import/' .. VAR.SINGULARITY_IMAGE_NAME)
 end

 inv.task('cleanup')
diff --git a/test/unit/tool_util/mulled/test_mulled_build.py b/test/unit/tool_util/mulled/test_mulled_build.py
```

Reply:
Ah! I looked at involucro's documentation but I found no explanation of what happens when composing multiple `.run()` but this makes sense!